crates/bitwarden/src/state/domain.rs

```diff
@@ -0,0 +1,188 @@
+use std::collections::HashMap;
+
+use std::str::FromStr;
+
+use chrono::DateTime;
+use chrono::Utc;
+use serde::Deserialize;
+use serde::Serialize;
+use uuid::Uuid;
+
+use bitwarden_api_api::models::{
+    CipherDetailsResponseModel, FolderResponseModel, ProfileResponseModel,
+};
+
+use crate::{
+    client::{auth_settings::AuthSettings, LoginMethod},
+    crypto::CipherString,
+    error::{Error, Result},
+};
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EnvironmentUrls {
+    pub base: Option<String>,
+    pub api: Option<String>,
+    pub identity: Option<String>,
+    pub web_vault: Option<String>,
+    pub icons: Option<String>,
+    pub notifications: Option<String>,
+    pub events: Option<String>,
+    pub key_connector: Option<String>,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AccountData {
+    pub keys: Option<Keys>,
+    pub profile: Option<Profile>,
+
+    pub ciphers: HashMap<Uuid, Cipher>,
+    pub folders: HashMap<Uuid, Folder>,
```

Comment:
I would like to have a generic way to assign data to an account, so that we don't necessarily fill this with 20 hashmaps, and everyone has to constantly modify it.

That would also allow the consumers to store data without needing to maintain a separate system for it.

Reply:
This struct wasn't in use in the last revision of the PR, it remained there from when the state was handled as a single block. I've deleted it now.

I think what you want to do can be done with `client .get_state_service(FOLDERS_SERVICE)`, by defining any service, it's just a type and a namespace. It's not exposed outside of the SDK but it can be done, we'd just need to avoid conflicting namespaces.